drivers/storage/portworx/component/portworx_basic.go

```diff
@@ -173,56 +179,106 @@ func (c *portworxBasic) createNamespace(namespace string) error {
 	return nil
 }
 
-func (c *portworxBasic) createRole(namespace string, ownerRef *metav1.OwnerReference) error {
-	return k8sutil.CreateOrUpdateRole(
-		c.k8sClient,
-		&rbacv1.Role{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:            PxRoleName,
-				Namespace:       namespace,
-				OwnerReferences: []metav1.OwnerReference{*ownerRef},
-			},
-			Rules: []rbacv1.PolicyRule{
-				{
-					APIGroups: []string{""},
-					Resources: []string{"secrets"},
-					Verbs:     []string{"get", "list", "create", "update", "patch", "delete"},
-				},
+func (c *portworxBasic) createRole(
+	cluster *corev1.StorageCluster,
+	namespace string,
+	ownerRef *metav1.OwnerReference,
+) error {
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      PxRoleName,
+			Namespace: namespace,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"secrets"},
+				Verbs:     []string{"get", "list", "create", "update", "patch", "delete"},
 			},
 		},
-		ownerRef,
+	}
+
+	if cluster.Namespace == namespace {
+		role.OwnerReferences = []metav1.OwnerReference{*ownerRef}
+		return k8sutil.CreateOrUpdateRole(c.k8sClient, role, ownerRef)
+	}
+
+	// Remove ownership information from the object as Kubernetes
+	// does not handle cross-namespace ownership. This code should
+	// be removed eventually when existing customers are upgraded.
+	existingRole := &rbacv1.Role{}
+	err := c.k8sClient.Get(
+		context.TODO(),
+		types.NamespacedName{
+			Name:      role.Name,
+			Namespace: role.Namespace,
+		},
+		existingRole,
 	)
+	if err == nil && len(existingRole.OwnerReferences) > 0 {
+		existingRole.OwnerReferences = nil
+		logrus.Debugf("Removing ownership from %s/%s Role", role.Namespace, role.Name)
+		if err := c.k8sClient.Update(context.TODO(), existingRole); err != nil {
```

Comment:
if we don't have the code here, later in this function k8sutil.CreateOrUpdateRole(c.k8sClient, role, nil) should do the same, right?

Reply:
The CreateOrUpdate* function will not update the object if we remove the ownerReference because of this line - 
```
if modified || len(role.OwnerReferences) > len(existingRole.OwnerReferences) {
```

Just trying to update the CreateOrUpdate* functions temporarily. We will remove the above code few releases down the line.

Reply:
Yes, I think it will work. we don't have to change it. Just to discuss the CreateOrUpdate function, I don't see other places using it, currently the function only merges owner references from exiting role, it does not handle removing owner references. Is there a use case for that?

Reply:
Generally we have never needed to remove owner references. Rather we only wanted to add ownerReferences. This is a one-off requirement because of the regression. That's why we are removing owner ref. Wanted to keep the CreateOrUpdateRole function consistent with other CreateOrUpdate* functions on that front. Hence kept the above update out of the CreateOrUpdateRole